microprofile/WFLY-14932_mp_reactive_messaging_kafka_api.adoc

```diff
@@ -0,0 +1,105 @@
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+The Kafka user API contains the following classes, the below shows what they allow you to access
+
+* https://github.com/smallrye/smallrye-reactive-messaging/pull/1272/files#diff-78e354d9794932a1d2b42b4c85153068ae53109160fe69218366a681d608f498[KafkaMetadataUtil] - contains utility methods to set `OutgoingKafkaRecordMetadata` in a Reactive Messaging `Message` to configure how messages are sent, and to read `IncomingKafkaRecordMetadata` from a `Message` received from Kafka to get information relating to Kafka. This utility class is present to wrap usage of `Metadata` which has made its way into the spec interfaces in the SmallRye Reactive Messaging API Maven module, while not yet part of the Microprofile Reactive Messaging specification.
+* https://github.com/smallrye/smallrye-reactive-messaging/pull/1272/files#diff-d372641705ed1e9167ae9ce27fa843860c02b1ec8460a6e38e9cc0e3e4e43b07[IncomingKafkaRecordMetadata] contains methods to read information relating to Kafka from an incoming `Message`. This information includes:
+** `key` - the key of the Kafka record represented by the `Message`.
+** `topic` - the Kafka topic the `Message` was received on.
+** `partition` - the Kafka partition the `Message` was received on.
+** `timestamp` - the timestamp of the `Message`.
+** `timeStampType` - the `org.apache.kafka.common.record.TimestampType` of the message.
+** `offset` - the Kafka offset of the Kafka record represented by the `Message`.
+** `headers` - An instance of `org.apache.kafka.common.header.Headers`. They have no meaning to the Kafka broker, which just forwards them on to the consumers. These can be thought of attachments/metadata on the `Message` to be used as additional information in the producer to consumer, and are dealt with on the application level. For example, a header could indicate which data centre was the origin of a message, and consumers are free to deal with this information or ignore it.
+* https://github.com/smallrye/smallrye-reactive-messaging/pull/1272/files#diff-097e0fa8873ba4f19a46c3229a185b93c2f12507785c8bcbb7b0a360ec66ee65[OutgoingKafkaRecordMetadata] contains a builder to set data to influence how Kafka sends a `Message`> This includes:
+** `key` - the key of the Kafka record represented by the `Message`.
+** `topic` - generally users should configure the Kafka topic for a stream via the application MicroProfile Config. However, in some cases they may want to specify the topic dynamically, for example depending on the contents of the data in the message that is being sent.
+** `partition` - the Kafka partition the `Message` was received on.
```

Comment:
just to clarify - this should probably say `the Kafka partition the Message will be sent to`?

Reply:
Done